### PR TITLE
Added collect and collect_into operator

### PR DESCRIPTION
**Pull request by Neo-Ciber94**
`collect` and `collect_into` are based on the methods `Iterator::collect` and `Iterator::collect_into` which allow to add all the items in an observable into a single collection line a `Vec<T>`.

This also a more generic [toArray](https://rxjs.dev/api/operators/toArray) operator.

### CODE REVIEW

**Comment by Neo-Ciber94** [comment]
Fixed clippy formatting error

File: CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +35,8 @@ This is a big refactor for rxRust, almost reimplement everything and many api wa
 - **operator**: add `throttle` operator.
 - **operator**: add `to_future` operator.
 - **operator**: add `to_stream` operator.
+- **operator**: add `collect` operator.
```
Comment (M-Adoo) [inline CHANGELOG.md:38]: Should not add in a released version,  add in `Unreleased` section is enough.

File: src/ops/collect.rs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+use crate::{
+  observable::{Observable, ObservableExt},
+  observer::Observer,
+};
+
+#[derive(Clone)]
+pub struct CollectOp<Item, S, C> {
+  source: S,
+  collection: C,
+  _marker: PhantomData<Item>,
```
Comment (M-Adoo) [inline src/ops/collect.rs:12]: This `_marker` unnecessary ,  you can bound `C` with `IntoIterator` add use its `Item` as `Extend<C::Item>`

File: src/ops/collect.rs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+impl<Item, S, C> CollectOp<Item, S, C> {
+  pub fn new(source: S, collection: C) -> Self {
+    CollectOp { source, collection, _marker: PhantomData }
+  }
+}
+
+impl<Item, Err, O, S, C> Observable<C, Err, O> for CollectOp<Item, S, C>
+where
+  C: Extend<Item>,
```
Comment (M-Adoo) [inline src/ops/collect.rs:23]: C: IntoIterator + Extend<<C as IntoIterator>::Item>

File: src/ops/collect.rs (hunk 2 above)
Comment (Neo-Ciber94) [reply src/ops/collect.rs:12]: Right, I overlooked that

**Review by M-Adoo — APPROVED** [review]

**Review by sologeek — APPROVED** [review]

**Pull request merged**